common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -772,11 +774,16 @@ def _load_items(self, course_entry, block_keys, depth=0, **kwargs):
         Load the definitions into each block if lazy is in kwargs and is False;
         otherwise, do not load the definitions - they'll be loaded later when needed.
         """
+        lazy = kwargs.pop('lazy', True)
+        should_cache_items = not lazy
+
         runtime = self._get_cache(course_entry.structure['_id'])
         if runtime is None:
-            lazy = kwargs.pop('lazy', True)
             runtime = self.create_runtime(course_entry, lazy)
             self._add_cache(course_entry.structure['_id'], runtime)
+            should_cache_items = True
+
+        if should_cache_items:
```

Comment:
Good question.  My change was an iterative, conservative change - changing the behavior only when `lazy=False`.

Now that you ask the question, I do wonder why we aren't always caching?  For some reason, the previous code chose to cache only when the runtime for the course wasn't cached.  Since there are many calls to `_load_items` from within split.py, there may be a good reason. (?).  Here's the precondition listed in `cache_items`:
```
Handles caching of items once inheritance and any other one time per course per fetch operations are done.
```